Feature: Finance read operations (#37)
aragon/sdk: javascript/modules/client/README.md

> ## Aragon JS SDK Client

Reviewer: Same as in the other review, why this change?

This is the markdown document title, if we start using # or ##, then everything else needs to be ### or more

### Loading DAO details

- **TODO**: Name, metadata, installed plugin list

### Loading DAO activity

- **TODO**: Transactions

### Loading DAO financial data

> Handles retrieving DAO asset balances using unique identifier.

Reviewer: ```suggestion
Handles retrieving DAO asset balances using the DAO address or its ENS domain.
```

```ts
import { Client } from "@aragon/sdk-client";
const client = new Client(context);

> const daoIdentifier = "0x1234..."; // unique identifier; dao name or address

Reviewer: ```suggestion
const daoIdentifier = "0x1234..."; // unique identifier; dao ENS domain or address
```

> const metadata = await client.methods.getDaoBalances(daoIdentifier);

Reviewer: Given that assets will be minted and indexed by Plugins themselves, I would expect the general-purpose `Client` class to receive the DAO address/ENS and an array of token contract addresses that we 100% want to receive (on top of the typical suspects like DAI, eth, USDC, ...)

> const metadata = await client.methods.getDaoBalances(daoIdentifier);

Reviewer: ```suggestion
const balances = await client.methods.getBalances(daoAddressOrEns);
```

console.log(metadata);
/* 
[{

>   "id": "0x1234...",

Reviewer: Why providing a DAO ID? the caller is passing it in

  "token": {

>     "id": "0x000...",

Reviewer: if that's the token address, then let's name it `address`. And in the case of Ether, I would set this field to `null`

    "name": "Ethereum (Canonical)",
    "symbol": "ETH",
    "decimals": "18"
  },
  "dao": {
    "id": "0x1234..."

>   },

Reviewer: Why this? is it the same as the `id` above?

>   "balance": "1100000000000000",

Reviewer: This should be a native JS `bigint` like the rest of the SDK does

  "lastUpdated": "1655983088"
},...] */
```

> Handles retrieving DAO transfer list using unique identifier.

Reviewer: ```suggestion
Retrieves the list of transfers made from and to a certain DAO.
```

```ts
import { Client } from "@aragon/sdk-client";
const client = new Client(context);

> const daoIdentifier = "0x1234..."; // unique identifier; dao name or address

Reviewer: The name could be self explanatory, indeed 

```suggestion
const daoAddressOrEns = "0x1234...";
```

> const metadata = await client.methods.getDaoTransfers(daoIdentifier);
> console.log(metadata);

Reviewer: ```suggestion
const transfers = await client.methods.getDaoTransfers(daoIdentifier);
console.log(transfers);
```

/*
{ 

>   "vaultDeposits": [{

Reviewer: ```suggestion
  "deposits": [{
```

>     "id": "0x1234...",

Reviewer: Why `id`?

    "dao": {
      "id": "0x1234...",

>     },

Reviewer: Why `dao: { id }`?

    "token": {

>       "id": "0x000...",

Reviewer: ```suggestion
      "address": "0x000...",
```

      "name": "Ethereum (Canonical)",
      "symbol": "ETH",
      "decimals": "18"
    },

>     "sender":"0x1234...",

Reviewer: ```suggestion
    "from":"0x1234...",
```

>     "amount": "1000000000000000",

Reviewer: Should be a native JS `bigint` (`BigInt(1234)`), noted as `1234n`
```suggestion
    "amount": 1000000000000000n,
```

    "reference": "",

>     "transaction": "0x1234...",

Reviewer: ```suggestion
    "transactionId": "0x1234...",
```

>     "createdAt": "1655983088"

Reviewer: A JavaScript native Date object should be returned

```suggestion
    "date": <Date>
```

  },...],

>   "vaultWithdraws": [{

Reviewer: ```suggestion
  "withdrawals": [{
```

    "id": "0x1234...",
    "dao": {
      "id": "0x1234...",

>     },

Reviewer: Why `id` and `dao { id }`?

    "token": {

>       "id": "0x000...",

Reviewer: In the case of Ether, should be null
```suggestion
      "address": "0x1234...", // null for Ether or the native token
```

      "name": "Ethereum (Canonical)",
      "symbol": "ETH",
      "decimals": "18"
    },

>     "sender":"0x1234...",

Reviewer: ```suggestion
    "from":"0x1234...",
```

>     "amount": "1000000000000000",

Reviewer: BigInt

    "reference": "",

>     "transaction": "0x1234...",

Reviewer: ```suggestion
    "transactionId": "0x1234...",
```

>     "createdAt": "1655983088"

Reviewer: JS Date

>   },...]

Reviewer: ```suggestion
  }, ...]
```

### Loading DAO financial data

Handles retrieving DAO asset balances using the DAO address or its ENS domain.

```ts
import { Client } from "@aragon/sdk-client";
const client = new Client(context);
const daoAddressOrEns = "0x1234..."; // unique identifier; dao ENS domain or address
const balances = await client.methods.getBalances(daoAddressOrEns);
console.log(balances);
/* 
[{
  "token": {
    "address": "0x000...",
    "name": "Ethereum (Canonical)",
    "symbol": "ETH",
    "decimals": "18"
  },

>   "balance": 1000000000000000n,

Reviewer: See suggestion below about the `AssetState` type

  "lastUpdate": <Date>
}, ...] */
```

Retrieves the list of transfers made from and to a certain DAO.

```ts
import { Client } from "@aragon/sdk-client";
const client = new Client(context);
const daoAddressOrEns = "0x1234...";

> const transfers = await client.methods.getDaoTransfers(daoAddressOrEns);

Reviewer: ```suggestion
const transfers = await client.methods.getTransfers(daoAddressOrEns);
```

console.log(transfers);
/*
{ 
  "deposits": [{
    "token": {
      "address": null,
      "name": "Ethereum (Canonical)",
      "symbol": "ETH",
      "decimals": "18"
    },
    "from":"0x1234...",
    "amount": 1000000000000000n,
    "reference": "",
    "transactionId": "0x1234...",
    "date": <Date>
  },...],
  "withdrawals": [{
    "token": {
      "address": null, // null for Ether or the native token
      "name": "Ethereum (Canonical)",
      "symbol": "ETH",
      "decimals": "18"
    },
    "to":"0x1234...",

>     "amount": 1000000000000000n,

Reviewer: Suggestion below about token transfer data type
